test/e2e/workflow/deployment_test.go

```diff
@@ -293,16 +293,22 @@ var _ = Describe("NetworkAddonsConfig", func() {
 				CreateConfig(gvk, p.configSpec)
 				CheckConfigCondition(gvk, ConditionAvailable, ConditionTrue, 15*time.Minute, CheckDoNotRepeat)
 
-				By("scraping the monitoring endpoint")
-				scrapedData, err := GetScrapedDataFromMonitoringEndpoint()
-				Expect(err).ToNot(HaveOccurred())
-
-				By("comparing the scraped Data to the expected metrics' values")
-				for metricName, expectedValue := range p.expectedMetricValueMap {
-					metricEntry := FindMetric(scrapedData, metricName)
-					Expect(metricEntry).ToNot(BeEmpty(), fmt.Sprintf("metric %s does not appear in endpoint scrape", metricName))
-					Expect(metricEntry).To(Equal(fmt.Sprintf("%s %s", metricName, expectedValue)), fmt.Sprintf("metric %s does not have the expected value %s", metricName, expectedValue))
-				}
+				Eventually(func() error {
+					By("scraping the monitoring endpoint")
+					scrapedData, err := GetScrapedDataFromMonitoringEndpoint()
+					Expect(err).ToNot(HaveOccurred())
+
+					By("comparing the scraped Data to the expected metrics' values")
+					for metricName, expectedValue := range p.expectedMetricValueMap {
+						metricEntry := FindMetric(scrapedData, metricName)
+						Expect(metricEntry).ToNot(BeEmpty(), fmt.Sprintf("metric %s does not appear in endpoint scrape", metricName))
+
+						if metricEntry != fmt.Sprintf("%s %s", metricName, expectedValue) {
+							return fmt.Errorf("metric %s does not have the expected value %s", metricName, expectedValue)
+						}
+					}
+					return nil
+				}, 2*time.Minute, 30*time.Second).Should(Succeed(), "Should scrape the correct metrics")
```

Comment:
I think we need the scraping inside the Eventually as well. 
the scrape "snapshots" the gauges state from the endpoint, so if the metrics weren't updated yet then we need to "reload" the snapshot.

Reply:
Then how come before the scraping was not in an `Eventually`?

Reply:
From some reason, it takes time for the status to be updated..
But it is strange now that I think of it. 
Let me investigate some more about it

Reply:
ah I see.

so, the status can be set to "available=true" by 2 controllers: the `networkaddonscondig_controller` which deploys and reconciles every 1minute, and the `pods_controller` which works every time one of the deployed pods changes. 
It it important to understand that both controllers change and update the status

Now currently I am "tracking" the monitored components and update the gauges every 1 minute in the `networkaddonscondig_controller`, but most times it is the `pods_controller` that actually notices and updates the status to Available=True. This means that we will only upate the gauges the next time the `networkaddonscondig_controller` launches (~1minute) 

What this means is that sometimes the gauges will take a bit to update. I don't think it matters that much, as the alert will fire only if one of the gauges turn to 0 for more than 5 minutes...
But it is a good question that you raised here.

I can also make sure to "track" the gauges on the other controller but I am not sure we want that..
@alonSadan, what do you think? Is the current suggested solution not good enough for us?

Reply:
Thanks for the explanation. I don't think it is necessary to track it  in both controllers, perhaps, since the pods controller is more sensitive to the pod changes, it make sense to move the tracking be only there?
but I agree it is not that crucial.

Reply:
I think the best place is `networkaddonscondig_controller`, as it is the main controller that decides whether the CR is deployed/removed.